### ISSUE TITLE
Editing Flow YAML in the UI

### ISSUE THREAD

**Issue by travjenkins** [estuary/ui#13]
![image](https://user-images.githubusercontent.com/270078/152004430-3d093870-6bbc-4676-aa92-fb9c5619123f.png)

We want to enable people to edit their configurations in both the CLI and the UI. This means that it would be great to have some of the niceties that are in IDEs (without just building a cloud IDE). We have been using [Monaco](https://microsoft.github.io/monaco-editor/) so far during dev. 

- Get schema validation visible to user
- Display YAML correctly (start with JSON first though)
- Can schema validation work for JSON?
- Displaying multiple files to match the structure in the directory

**Comment by travjenkins** [comment]
This is pretty far along now (though not a lot of code is written for it).

We are waiting on a more solid response from the discovery endpoint. I know Alex is working hard on getting that all coded up and merged (we already reviewed it manually with him and it looks good).

Once we have that response we can decide how we'll display multiple files, maybe builds forms on top, etc.

**Comment by travjenkins** [comment]
This is nearing being done. We can now select the file we want to edit, edit it, and have those changes saved in the local store.

I'd like to add is some schema validation before closing.

**Comment by travjenkins** [comment]
This is done. The editing isn't easy - but it is there. Will make a new issue for adding schema validation and other nice features.
